libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -73,8 +73,10 @@ class YulUtilFunctions
 
 	/// @returns a function that copies raw bytes of dynamic length from calldata
 	/// or memory to memory.
-	/// Pads with zeros and might write more than exactly length.
-	std::string copyToMemoryFunction(bool _fromCalldata);
+	/// @params _unsafeCleanup: If true, pads with zeros and might write more
+	///         than the exact length, else only up until the exact length will
+	///         be written
```

Comment:
Actually, does it ever **not** write more than necessary? I think that even if the length is a multiple of 32, it does the cleanup even though it's not necessary.

```suggestion
	/// @params _unsafeCleanup: If true, overwrites exactly 32 bytes past the end of the array with zeros.
	///         WARNING: The cleaned area always extends beyond the padding resulting from 32-byte alignment.
	///         If you use this value, you must ensure on your own that the data past the end of the array is
	///         preserved or is fine to overwrite.
```